openshift/smoke-test.yaml

```diff
@@ -0,0 +1,90 @@
+---
+kind: Template
+apiVersion: v1
+metadata:
+  name: openshift-web-console-smoke-test-template
+  annotations:
+    description: >
+      Runs smoke tests in a headless browser against the web console at a provided URL.
+
+      For more information about using this template, see
+      https://github.com/benjaminapetersen/origin-web-console-smoke-test/blob/master/README.md
+objects:
+- apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: openshift-web-console-smoke-test-deployment
+    labels:
+      app: openshift-web-console-smoke-test
+  spec:
+    # replica set
+    replicas: 1
+    # selector = how the deployment finds pods to manage
+    selector:
+      matchLabels:
+        app: openshift-web-console-smoke-test
+    template:
+      metadata:
+        labels:
+          app: openshift-web-console-smoke-test
+      spec:
+        containers:
+        - name: openshift-web-console-smoke-test
+          image: benjaminapetersen/openshift-web-console-smoke-test:latest
+          imagePullPolicy: IfNotPresent
+          env:
+          - name: CONSOLE_URL
+            value: ${CONSOLE_URL}
+          - name: CONSOLE_USER
+            value: ${CONSOLE_USER}
+          - name: CONSOLE_PASSWORD
+            value: ${CONSOLE_PASSWORD}
+          - name: TEST_INTERVAL_MINUTES
+            value: ${TEST_INTERVAL_MINUTES}
+          ports:
+          - containerPort: 3000
+          restartPolicy: Always
+          volumeMounts:
+            - mountPath: /var/tls-certs
+              name: web-console-smoke-test-serving-cert
+              readOnly: true
+        volumes:
+        - name: web-console-smoke-test-serving-cert
+          secret:
+            defaultMode: 400
+            secretName: web-console-smoke-test-serving-cert
+
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: openshift-web-console-smoke-test-service
+    annotations:
+      service.alpha.openshift.io/serving-cert-secret-name: web-console-smoke-test-serving-cert
+        # TODO:
+        # don't we need these annotations, just like the web-console?
+        # https://github.com/openshift/origin/blob/master/install/origin-web-console/console-template.yaml#L127
+        # prometheus.io/scrape: "true"
+        # prometheus.io/scheme: https
+  spec:
+    ports:
+    - name: web
+      protocol: TCP
+      port: 443
+      targetPort: 3000
+      nodePort: 0
+    selector:
+      app: openshift-web-console-smoke-test
+    type: ClusterIP
+    sessionAffinity: None
+  status:
+    loadBalancer: {}
+parameters:
+- name: CONSOLE_URL
+  description: The location of the web console, in the format of https://<machine-ip>:8443
+  value: https://192.168.64.3:8443
```

Comment:
Not sure if we want to have a default value here. I'd rather have this parameter without it and set to be mandatory to provide, by adding `required: true`

Reply:
Agree, lemme update that.